addons/web/controllers/dataset.py

```diff
@@ -27,11 +29,17 @@ def _call_kw_readonly(self):
 
     @http.route(['/web/dataset/call_kw', '/web/dataset/call_kw/<path:path>'], type='jsonrpc', auth="user", readonly=_call_kw_readonly)
     def call_kw(self, model, method, args, kwargs, path=None):
+        rpc_path = f'{model}/{method}'
```

Comment:
You should put this inside `call_kw`

Reply:
Doing so would break the test  https://github.com/odoo/odoo/pull/195886/files#diff-89dcb3f3760bc6db1c7f8cdbfeb7abb6e5c5561190270e114e7b03bad1be7131R78-R88

It would log
```
POST /web/dataset/call_kw/res.users/read#res.users/read
```

Reply:
and when I'm inside `service/model.py` I have no access to `request` anymore (`borrow_request`) so I cannot move the conditional `if not request.httprequest.path.endswith(...)` down there

Reply:
`path` was ignored, so you could just update `request.path` instead and remove `path` from it.
You should raise an exception if the path is set and does not match `rpc_path`.

Adding the fragment in `call_kw` will make sure that all calls have their arguments logged.

Reply:
I am not sure to understand. Can you take 20 minutes and test what you suggest on your side, feel free to push a commit on top.

Reply:
Ok so commit is at https://github.com/odoo/odoo/pull/195886/commits/92674f8d40aaab5e54a996774a41d4d698d22b9c and I don't like it for two reasons:

1. raising a BadRequest makes an unwelcome backward incompatibility, we didn't specified what "path" was so people were free to use it however they liked, and I don't think it is a good idea to starting responding with 400 because now we decided to give it purpose.
2. for the webclient and /web/dataset/call_kw (that is the main rpc entry point) it duplicates the rpc_path `/web/dataset/call_kw/res.users/read#res.users/read`

Thank you though! I thought you had a solution to solve (2).